toolchain/check/testdata/array/array_in_place.carbon

```diff
@@ -23,11 +23,13 @@ fn G() {
 // CHECK:STDOUT:   %.loc10_28: i32 = int_literal 2
 // CHECK:STDOUT:   %.loc10_29: type = array_type %.loc10_28, (i32, i32, i32)
 // CHECK:STDOUT:   %v: ref [(i32, i32, i32); 2] = var "v"
+// CHECK:STDOUT:   %F.ref.loc10_34 = name_reference_untyped "F", package.%.loc7
```

Comment:
For package references, this doesn't really seem to be working the way I would expect.

The `package.%.loc7` is much less friendly than `@F`, and we don't use the `F.ref.loc10_34` below in the `call` instruction.

Are either of these worth fixing in this iteration?

Reply:
Yeah, I think there's still work to be done here:

- We should name the package-scope values after the functions they represent.
- Calls should reference the node they're using to name the function, rather than referring to the function directly.
- Name qualifiers like `N.M.F` should reference the various name components in some way. This PR only adds SemIR for `N`. This is a prerequisite for the previous bullet because we currently have no node representing the callee in `N.M.F()`.

I think the first and third changes are best done in separate PRs, and the second one depends on the third so it makes most sense to defer that too.

Reply:
PR for item 1 landed before this PR :)